### PR TITLE
Add method validation tests and fixes detected by those tests

### PR DESCRIPTION
**Pull request by seratch**
###  Summary

This pull request adds automated tests verifying if all the fields are supported in the Kotlin DSL and the setter names are consistent with underlying Java SDK and Slack APIs.

Also, I've fixed some inconsistency issues and missing field supports detected by the tests.

### CODE REVIEW

File: slack-api-model-kotlin-extension/src/main/kotlin/com/slack/api/model/kotlin_extension/block/composition/OptionGroupObjectBuilder.kt [hunk 1]
```diff
@@ -26,6 +27,10 @@ class OptionGroupObjectBuilder private constructor(
         label = PlainTextObject(text, emoji)
     }
 
+    fun options(options: List<OptionObject>) {
```
Comment (seratch) [inline slack-api-model-kotlin-extension/src/main/kotlin/com/slack/api/model/kotlin_extension/block/composition/OptionGroupObjectBuilder.kt:30]: The tests detected lack of this method. Most people won't use this but I would like to have this for consistency.

File: slack-api-model-kotlin-extension/src/main/kotlin/com/slack/api/model/kotlin_extension/block/element/RichTextSectionElementChannelBuilder.kt [hunk 2]
```diff
@@ -7,7 +7,7 @@ import com.slack.api.model.kotlin_extension.block.element.container.SingleRichTe
 import com.slack.api.model.kotlin_extension.block.element.dsl.RichTextStyleDsl
 
 @BlockLayoutBuilder
-class RichTextElementChannelBuilder private constructor(
+class RichTextSectionElementChannelBuilder private constructor(
```
Comment (seratch) [inline slack-api-model-kotlin-extension/src/main/kotlin/com/slack/api/model/kotlin_extension/block/element/RichTextSectionElementChannelBuilder.kt:10]: Renamed all builder classes that construct `RichTextSectionElement.*`

File: slack-api-model-kotlin-extension/src/test/kotlin/test_locally/block/validation/ValidationUtil.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package test_locally.block.validation
+
+import java.lang.reflect.Modifier
+import kotlin.test.assertTrue
+
+object ValidationUtil {
+
+    fun validateMethodNames(target: Class<*>, builder: Class<*>) {
+
+        val expectedNames = target
```
Comment (seratch) [inline slack-api-model-kotlin-extension/src/test/kotlin/test_locally/block/validation/ValidationUtil.kt:10]: I think using Java reflection is fine but if you have an alternative idea to implement this test utility, I'd love to know the idea. That said, that's a nice-to-have thing. We can discuss it separately.

Comment (emanguy) [reply slack-api-model-kotlin-extension/src/test/kotlin/test_locally/block/validation/ValidationUtil.kt:10]: I'm gonna try a method where we look up classes which have the `@BlockLayoutBuilder` annotation to automatically pull in the builder classes for comparison. If it takes too long or is finnicky I'll give it up.

File: slack-api-model-kotlin-extension/src/main/kotlin/com/slack/api/model/kotlin_extension/block/composition/OptionGroupObjectBuilder.kt (hunk 1 above)
Comment (emanguy) [reply slack-api-model-kotlin-extension/src/main/kotlin/com/slack/api/model/kotlin_extension/block/composition/OptionGroupObjectBuilder.kt:30]: After merge I'll refactor to use the OptionObjectBuilder

File: slack-api-model-kotlin-extension/src/main/kotlin/com/slack/api/model/kotlin_extension/block/composition/ConfirmationDialogObjectBuilder.kt [hunk 4]
```diff
@@ -52,12 +53,25 @@ class ConfirmationDialogObjectBuilder private constructor(
...
 
+    /**
+     * Defines the color scheme applied to the confirm button.
+     * A value of danger will display the button with a red background on desktop, or red text on mobile.
+     * A value of primary will display the button with a green background on desktop, or blue text on mobile.
+     * If this field is not provided, the default value will be primary.
+     *
+     * @see <a href="https://api.slack.com/reference/block-kit/composition-objects#confirm">Confirmation dialog object documentation</a>
+     */
+    fun style(style: String) {
```
Comment (emanguy) [inline slack-api-model-kotlin-extension/src/main/kotlin/com/slack/api/model/kotlin_extension/block/composition/ConfirmationDialogObjectBuilder.kt:64]: I'm also going to add an overload for this which uses `ButtonStyle`

**Review by emanguy — APPROVED** [review]
Merging in your changes now.

**Pull request merged**